### DISCUSSION TITLE
Redis url for redisraft

### DISCUSSION THREAD

**Discussion by diseq** [juicedata/juicefs#854] [Q&A] [answered]
I have redisraft deployed and using it with juicefs. Using a single instance works fine.
However, listing them all in a the connectionstring does not work. (in case of a fallback scenario)

Is there a different format using raft?

```
/usr/sbin/juicefs mount redis://10.1.248.149,10.1.129.75:6379/1 /mnt/jfs

juicefs[1007710]: 2021/09/21 14:46:55.542089 juicefs[1007710] <INFO>: Meta address: redis://10.1.248.149,10.1.129.75:6379/1
juicefs[1007710]: redis: 2021/09/21 14:46:55 sentinel.go:471: sentinel: GetMasterAddrByName master="10.1.248.149" failed: MOVED 0 10.1.248.149:6379
juicefs[1007710]: 2021/09/21 14:46:55.546278 juicefs[1007710] <WARNING>: parse info: redis: all sentinels are unreachable
juicefs[1007710]: redis: 2021/09/21 14:46:55 sentinel.go:471: sentinel: GetMasterAddrByName master="10.1.248.149" failed: MOVED 0 10.1.248.149:6379
juicefs[1007710]: 2021/09/21 14:46:55.548456 juicefs[1007710] <FATAL>: load setting: redis: all sentinels are unreachable
```

**Comment by xiaogaozi** [comment, marked as answer]
Hi @diseq, you should specify only Raft leader in the metadata URL, e.g. `redis://10.1.248.149:6379/1`.

BTW, We have not officially supported RedisRaft, please use with caution.

**Reply by diseq** [reply]
Got it. Specifying just one node works like a charm.

I just want to make sure in case of failure it automatically switches over. 
Thought I could avoid a loadbalancer/vip in the front but seems the way to go.

**Reply by xiaogaozi** [reply]
JuiceFS relies on [go-redis](https://github.com/go-redis/redis), you need to confirm whether go-redis supports RedisRaft.

**Reply by deanpatel2** [reply]
Hello, is this still the format the meta URL should be passed in when using Sentinel?

Docs are telling me `redis[s]://[[USER]:PASSWORD@]MASTER_NAME,SENTINEL_ADDR[,SENTINEL_ADDR]:SENTINEL_PORT[/DB]` but when I do this I'm getting the same error logs above. 

```
redis: 2024/09/05 19:30:22 sentinel.go:537: sentinel: GetMasterAddrByName master="redis-sentinel" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
2024/09/05 19:30:22.316619 juicefs[18] <FATAL>: redis: all sentinels specified in configuration are unreachable [main.go:31]
```

Even though my master is named `redis-sentinel` and I am passing the password in the URL and the Sentinel password as `SENTINEL_PASSWORD` environment variable. I have no problem connecting to the Sentinel from a different service, just with JuiceFS so I believe it is a formatting issue.

@xiaogaozi any help would be much appreciated 🙇

**Reply by xiaogaozi** [reply]
@deanpatel2 Hi, if you are sure you can reproduce the issue, please create an issue with a detailed description of the steps to reproduce the issue and information about your environment (e.g., operating system, Redis version, JuiceFS client version).

**Reply by deanpatel2** [reply]
@xiaogaozi I just made an issue https://github.com/juicedata/juicefs/issues/5149. Thank you for your help 🙏

**Reply by deanpatel2** [reply]
@xiaogaozi can you help?

**Reply by xiaogaozi** [reply]
@deanpatel2 Hi, my colleague has already replied in the issue.
